charts/nri-metadata-injection/templates/deployment.yaml

```diff
@@ -20,9 +20,6 @@ spec:
       labels:
         {{- include "nri-metadata-injection.labels" . | nindent 8 }}
     spec:
-      {{- if (and (not .Values.customTLSCertificate) (not .Values.certManager.enabled)) }}
```

Comment:
Why did we need the service account?

Reply:
I did not find any reason why we were adding such (broad) permissions to it

Reply:
@douglascamata do you remember?

Reply:
I don't know. It was there since the beginning of git history. Maybe in the past certificates were generated as part of this deployment? I went trough the k8s-metadata-injection code and I don't think it needs the service account. I can remove it in separate commit though.

Reply:
I'm not sure as well about the need for the service account in this deployment. It's completely stateless, it doesn't do any call to the api server... 🤔